File: packages/lib-angular/projects/avatars/src/lib/avatars/avatar-ring.utils.ts
import { hashCode, getRandomColor } from "../avatar.utils";

export const SIZE = 90;
export const COLORS = 5;

export function generateColors(colors: string[], name: string) {
  const numFromName = hashCode(name);
  const range = colors && colors.length;
  const colorsShuffle = Array.from({ length: COLORS }, (_, i) =>
    getRandomColor(numFromName + i, colors, range)
  );
  const colorsList = [];
  colorsList[0] = colorsShuffle[0];
  colorsList[1] = colorsShuffle[1];
  colorsList[2] = colorsShuffle[1];
  colorsList[3] = colorsShuffle[2];
  colorsList[4] = colorsShuffle[2];
  colorsList[5] = colorsShuffle[3];
  colorsList[6] = colorsShuffle[3];
  colorsList[7] = colorsShuffle[0];
  colorsList[8] = colorsShuffle[4];

  return colorsList;
}

File: packages/lib-leptos/src/components/avatar_ring_utils.rs
use crate::components::avatar_utils::*;

pub const SIZE: usize = 90;
pub const COLORS: usize = 5;

pub fn generate_colors(name: &str, colors: Vec<&str>) -> Vec<String> {
    let num_from_name = hash_code(name);
    let range = colors.len();

    let mut generated_colors: Vec<String> = vec![];

    for i in 0..COLORS {
        generated_colors.push(get_random_color(
            num_from_name + i as i64,
            &colors,
            range as i64,
        ));
    }

    let mut oddly_generated_colors: Vec<String> = vec![];

    oddly_generated_colors.push(generated_colors[0].clone());
    oddly_generated_colors.push(generated_colors[1].clone());
    oddly_generated_colors.push(generated_colors[1].clone());
    oddly_generated_colors.push(generated_colors[2].clone());
    oddly_generated_colors.push(generated_colors[2].clone());
    oddly_generated_colors.push(generated_colors[3].clone());
    oddly_generated_colors.push(generated_colors[3].clone());
    oddly_generated_colors.push(generated_colors[0].clone());
    oddly_generated_colors.push(generated_colors[4].clone());

    oddly_generated_colors
}
